applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/adjoint_nodal_reaction_response_function.cpp

```diff
@@ -278,16 +274,41 @@ namespace Kratos
 
             if( rAdjointCondition.Id() == ng_cond_i.Id() )
             {
-                Vector filter_vector;
                 ProcessInfo process_info = rProcessInfo;
-                this->ConstructFilterVector(rAdjointCondition, filter_vector, process_info);
-                noalias(rSensitivityGradient) = prod(rSensitivityMatrix, filter_vector);
+                auto dof_index = this->GetDofIndex(ng_cond_i, process_info);
+                rSensitivityGradient = -1.0 * (this->GetColumnCopy(rSensitivityMatrix, dof_index));
             }
         }
 
         KRATOS_CATCH("");
     }
 
+    Vector AdjointNodalReactionResponseFunction::GetColumnCopy(Matrix m, size_t i)
```

Comment:
`const Matrix&` Otherwise you make the copy of the matrix here and then you could use the existing functionality.